discovery_engine_core/core/src/engine.rs

```diff
@@ -96,6 +96,13 @@ use crate::{
     },
 };
 
+#[cfg(feature = "storage")]
+type BoxedStorage = Box<
+    dyn Storage<StorageError = sqlite::StorageError, FeedScopeError = sqlite::FeedScopeError>
+        + Send
+        + Sync,
+>;
```

Comment:
I have thought about it some more.

To avoid verbosity/accidental complexity we probably would want to use roughly the same error type (or same type per scope) for all backends, e.g. sqlite and postgres. Naturally that won't work for the error variant for the sqlite error but lets ignore that for a moment.

The reason I'm saying this is because if we don't do so we need to be generic over the error in any place we want to potentially match over it... which would be quite annoying.

If we use the same error type(s) anyway we could just not make the traits generic over their error type. I mean the trait is only used internally so why not.

I think any db specific error we expect as part of the domain logic we should convert to it's own error variant, any error left is a but we anyway can only report back in a way which is in the end "somehow" reported to us (i.e. report to team blue which then logs an error we then might be reported to us).

Which means we could just box any db specific error not handled by db specific code.

Reply:
The associated error type has been removed. We now have a simple storage error type/enum with currently just `Database` as the only variant for both traits. Any database error will be converted into a generic error. I think github didn't pick up the latest changes. please have a look again